### PR TITLE
Add Prometheus sink

### PR DESCRIPTION
**Pull request by golfdish**
Merging into new branch `promsink` (off of master@v0.4.5) for development release.

`promauto` is cautioned against because panics can happen at runtime if metric registration fails, but since creation/registration is abstracted behind the stats library, this is unavoidable anyway.

### CODE REVIEW

File: prometheus_sink.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+type prometheusSink struct {
+	counters   map[string]prometheus.Counter
+	gauges     map[string]prometheus.Gauge
+	histograms map[string]prometheus.Histogram
+}
+
+// NewLoggingSink returns a Sink that flushes stats to os.StdErr.
+func NewPrometheusSink() Sink {
+	http.Handle("/metrics", promhttp.Handler())
+	go http.ListenAndServe(":2112", nil)
```
Comment (lorenmh) [inline prometheus_sink.go:20]: any reason for `":2112"` (ie vs ":9090")?

Comment (golfdish) [reply prometheus_sink.go]: copypasta

**Review by lorenmh — APPROVED** [review]
lgtm

**Comment by zirain** [comment]
@lorenmh @golfdish what's current status of prom sink? will it be merged to master? if so, when will it happen?

**Comment by golfdish** [comment]
Loren no longer works at Lyft. You may want to ask @njo or @kirtia1 about this, I've since paged out my context for it.

**Pull request merged**